### ISSUE TITLE
Syncing two clients returns 410 after server state is deleted

### ISSUE THREAD

**Issue by Akendo** [GothenburgBitFactory/taskchampion-sync-server#47]
Hi everyone,

I'm in the process of migrating my taskd to taskchampion-sync-serve. I've set up the server as describe in the README. Also, I have two clients that I migrated to task 3. 

When I try to sync them, I get the following error message:
```
480 ~/s/taskwarrior-3 (develop|✔) $ ./build/src/task sync
Failed to synchronize with server: Server Error: http://task.internal:8080/v1/client/get-child-version/fd16fa03-30d9-4e39-a7e0-5fdddc2d6a92 responded with 410 Gone
```

The different should be only a single task. Any suggest what the problem might be?

**Comment by djmitche** [comment]
Hm, that error occurs when a version that used to exist on the server, as remembered by the client, no longer exists. Perhaps at some point you reset the state of the server? Because the client "remembers" the last version it uploaded, doing so is a bit problematic!

Do the two clients currently have the same set of tasks, or is one of the two "correct"? If not, that will need some manual intervention. Once that's done, I'd suggest `task export` the correct data and start from scratch:
 - reset the server state (just delete its DB file)
 - reset both client states (delete their DB files)
 - on one client, run `task import` and then `task sync`
 - on the other client, run `task sync`
At this point, you should see the same tasks on both clients.

There _is_ a bug here, without which recovery would be a little easier. Currently, if you take a replica that has already been syncing to some other server for some time (and thus remembers the last version it uploaded), then delete the server state, the server should allow simply uploading the new version, as long as the server state is empty. However, as you've seen right now it replies with "410 Gone" which the client interprets as "the child version existed and has been deleted" and reports an error -- this might happen on a client that hadn't been synced in months or years, for example. Returning a "404 Not Found" would be better, as the client interprets that response as "there's no child version, go ahead and upload one".

Anyway, please try the above steps and once you're back up and running, I can take a look at fixing the bug. The fix isn't hard, but I'll want to update the protocol specification as well to ensure other implementations don't make the same mistake.

**Comment by djmitche** [comment]
@Akendo any luck?

**Comment by djmitche** [comment]
(I still need to finish the docs and make a PR for that)

**Comment by Akendo** [comment]
Thank you very much, I was able to make it running by deleting the container with the service that it was running it. Hence, resetting the initial server state was the key.

However, I feel like we need to make the service more resilience to deal with such situations. It means that one of the clients created an inconsistent state and I feel like that's something that's going to happen once in a while, and I feel like resetting the database seems like not the best way to deal with this. 

Not sure how I can do this, I'm fairly limited with time, but testing is something I can do for sure. 

Still, I appreciated it very much. Keep up the good work!
